packages/typescriptlang-org/src/lib/useLocalStorage.ts

```diff
@@ -0,0 +1,12 @@
+import { useState } from "react"
+
+export const useLocalStorage = <State extends string>(key: string, initialValue: State) => {
+    const [value, setValue] = useState(globalThis.localStorage?.getItem(key) ?? initialValue)
```

Comment:
Careful, you have to [verify local storage](https://github.com/microsoft/TypeScript-website/blob/1c68e217fd03e5e02e6a7685457fca5a42b23f34/packages/typescriptlang-org/src/components/ShowExamples.tsx#L87-L95) exists first:

```
let hasLocalStorage = false
    try {
      hasLocalStorage = typeof localStorage !== `undefined`
    } catch (error) { }

    if (hasLocalStorage) {
      const examplesFromLS = localStorage.getItem("examples-seen") || "{}"
      seenExamples = JSON.parse(examplesFromLS)
    }
```

Otherwise usage will throw if someone has high enough privacy settings